app/src/main/java/com/arctouch/codechallenge/view/home/HomeFragment.kt

```diff
@@ -0,0 +1,62 @@
+package com.arctouch.codechallenge.view.home
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.navigation.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.util.InjectorUtils
+import com.arctouch.codechallenge.viewmodel.HomeViewModel
+import kotlinx.android.synthetic.main.fragment_home.*
+
+
+class HomeFragment : Fragment() {
+
+    companion object {
+        fun newInstance() = HomeFragment()
+    }
+
+    private lateinit var viewModel: HomeViewModel
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_home, container, false)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        val factory = InjectorUtils().provideViewModelFactory()
+        viewModel = ViewModelProviders.of(this, factory).get(HomeViewModel::class.java)
+        setupHome()
+        observe()
+    }
+
+    private fun observe(){
+        viewModel.getUpcomingMovies().observe(this, Observer { movies ->
+           homePagedAdapter.submitList(movies)
+           progressBar.visibility = View.GONE
+        })
+    }
+
+    private fun setupHome(){
+        val linearLayoutManager = LinearLayoutManager(context)
+        recyclerView.layoutManager = linearLayoutManager
+        recyclerView.adapter = homePagedAdapter
+    }
+
+    private val homePagedAdapter = HomePagedAdapter{ movie ->
+        val bundle = Bundle()
```

Comment:
Since you are using the  ktx dependency in your project, why not applying it here?

Reply:
@arctouch-matheusfaleiro my mistake
the correct would be: val bundle = bundleOf ("movie_id" to movie.id.toLong (),
"movie_poster" to movie.posterPath,
"movie_backdrop" to movie.backdropPath)